etc/deploy/restart_with_tls.sh

```diff
@@ -18,44 +20,40 @@ while true; do
   esac
 done
 
-# Turn down old pachd deployment, so that new (TLS-enabled) pachd doesn't try to connect to old, non-TLS pods
-# I'm not sure why this is necessary -- pachd should communicate with itself via an unencrypted, internal port
-# Empirically, though, the new pachd pod crashloops if I don't do this (2018/6/22)
-kubectl get deploy/pachd -o json | jq '.spec.replicas = 0' | kubectl apply -f -
+echo "Turning down any existing pachyderm cluster..."
+if pachctl version --timeout=5s; then
+  # Turn down old pachd deployment, so that new (TLS-enabled) pachd doesn't try
+  # to connect to old, non-TLS pods I'm not sure why this is necessary -- pachd
+  # should communicate with itself via an unencrypted, internal port.
+  #
+  # Empirically, though, new pachd pods crashloop if I don't do this (2018/6/22)
```

Comment:
This sounds pretty fishy. Could you file a ticket for it, just so we don't lose track of this issue?

Reply:
Done: https://github.com/pachyderm/pachyderm/issues/4173